src/lib/esnext.iterator.d.ts

```diff
@@ -0,0 +1,125 @@
+         * @param initialValue If initialValue is specified, it is used as the initial value to start the accumulation. The first call to the callbackfn function provides this value as an argument instead of a value from the iterator.
+         */
+        reduce<U>(callbackfn: (previousValue: U, currentValue: T, currentIndex: number) => U, initialValue: U): U;
+
+        /**
+         * Creates a new array from the values yielded by this iterator.
+         */
+        toArray(): Array<T>;
+
+        /**
+         * Performs the specified action for each element in the iterator.
+         * @param callbackfn  A function that accepts up to two arguments. forEach calls the callbackfn function one time for each element in the iterator.
+         */
+        forEach(callbackfn: (value: T, index: number) => void): void;
+
+        /**
+         * Determines whether the specified callback function returns true for any element of this iterator.
+         * @param predicate A function that accepts up to two arguments. The some method calls
+         * the predicate function for each element in this iterator until the predicate returns a value
+         * true, or until the end of the iterator.
+         */
+        some(predicate: (value: T, index: number) => unknown): boolean;
+
+        /**
+         * Determines whether all the members of this iterator satisfy the specified test.
+         * @param predicate A function that accepts up to two arguments. The every method calls
+         * the predicate function for each element in this iterator until the predicate returns
+         * false, or until the end of this iterator.
+         */
+        every(predicate: (value: T, index: number) => unknown): boolean;
+
+        /**
+         * Returns the value of the first element in this iterator where predicate is true, and undefined
+         * otherwise.
+         * @param predicate find calls predicate once for each element of this iterator, in
+         * order, until it finds one where predicate returns true. If such an element is found, find
+         * immediately returns that element value. Otherwise, find returns undefined.
+         */
+        find<S extends T>(predicate: (value: T, index: number) => value is S): S | undefined;
+        find(predicate: (value: T, index: number) => unknown): T | undefined;
+
+        readonly [Symbol.toStringTag]: string;
+    }
+
+    // Global `IteratorConstructor` interface that can be augmented by polyfills
+    interface IteratorConstructor extends BuiltinIteratorConstructor {
+        /**
+         * Creates a native iterator from an iterator or iterable object.
+         * Returns its input if the input already inherits from the built-in Iterator class.
+         * @param value An iterator or iterable object to convert a native iterator.
+         */
+        from<T>(value: Iterator<T> | Iterable<T>): BuiltinIterator<T>;

```

Comment:
Should we restrict this to only allow `Iterator<U, unknown, undefined> | Iterable<U, unknown, undefined>`, since `from` is going to call `next()` on the result without passing an argument?

Reply:
I've added this restriction for now, since it seems useful, though we don't prevent you from using `.map` on a `Generator<string, boolean, number>`, even though the generator expects to receive `number` in calls to `next()`, and not `undefined`.